### ISSUE TITLE
running goal 'test' with compilationLevel 'ADVANCED' removes Assertions

### ISSUE THREAD

**Issue by FrankHossfeld** [Vertispan/j2clmavenplugin#38]
To reproduce the error clone f.e.: [gwt-aria module](https://github.com/gwtproject/gwt-aria) and run `j2cl:test`inside the `gwt-aria-j2cl-test`-module. The method `testSetDefaultValue_noSet` inside the `AttributeJ2clTest`-class will fail.

Background:
Create a `PrimitiveValueAttribute` with no default value. Now try to set the default value using `setDefaultValue`. The expected result is an AssertionError. 
 Using **compilationLevel** = 'BUNDLE', the test case runs as expected. Running the test with **compilationLevel** = 'ADVANCED' fails.

It looks like Assertions gets removed when running with **compilationLevel** = 'ADVANCED'

**Comment by niloc132** [comment]
It looks like this comes from `com.google.javascript.jscomp.J2clAssertRemovalPass` in closure. There is no command-line way to disable this, and it isn't wired into the bazel rules as far as I can tell, but we can wire up our own flag to modify this internally: `com.google.javascript.jscomp.CompilerOptions#setRemoveJ2clAsserts`.
